pathfinder/tests/test_blockchain_integration.py

```diff
@@ -0,0 +1,140 @@
+        (1, clients[1], 40, 10, 0.0008, 2, clients[2], 130, 100, 0.0012),  # capacities 130 --  40
+        (2, clients[2], 90, 10, 0.0007, 3, clients[3], 10, 0, 0.0010),  # capacities  80 --  10
+        (3, clients[3],  50, 20, 0.0011, 4, clients[4], 50, 20, 0.0011),  # capacities  50 --  50
+        (0, clients[0],  40, 40, 0.0015, 2, clients[2], 80, 0, 0.0025),  # capacities   0 -- 120
+        (1, clients[1],  30, 10, 0.0100, 4, clients[4], 40, 15, 0.0018),  # capacities  35 --  35
+        (5, clients[5], 500, 900, 0.0030, 6, clients[6], 750, 950, 0.0040),  # capacities 550 --
+        #  700
+    ]
+
+    blockchain_listener.required_confirmations = 1
+    token_network = pathfinding_service.token_networks[network_address]
+    pathfinding_service.token_network_listener.start()
+
+    for (
+        p1_index,
+        p1,
+        p1_deposit,
+        p1_transferred_amount,
+        p1_fee,
+        p2_index,
+        p2,
+        p2_deposit,
+        p2_transferred_amount,
+        p2_fee
+    ) in channel_descriptions:
+
+        p2.open_channel(p1.address)
+        p1.deposit_to_channel(p2.address, p1_deposit)
+        p2.deposit_to_channel(p1.address, p2_deposit)
+        ethereum_tester.mine_blocks(1)
+        gevent.sleep(0)
+
+    # there should be as many open channels as described
+    assert len(token_network.channel_id_to_addresses.keys()) == len(channel_descriptions)
+
+    # check that deposits got registered
+    for channel_id, (
+        p1_index,
+        p1,
+        p1_deposit,
+        p1_transferred_amount,
+        p1_fee,
+        p2_index,
+        p2,
+        p2_deposit,
+        p2_transferred_amount,
+        p2_fee
+    ) in enumerate(channel_descriptions):
+        p1, p2 = token_network.channel_id_to_addresses[channel_id+1]
+        view1 = token_network.G[p1][p2]['view']
+        view2 = token_network.G[p2][p1]['view']
+
+        assert view1.deposit == p1_deposit
+        assert view2.deposit == p2_deposit
+
+    # check pathfinding
+    paths = token_network.get_paths(clients[0].address, clients[3].address, 10, 5)
+    assert len(paths) == 3
+    assert paths[0]['path'] == [clients[0].address, clients[2].address, clients[3].address]
+    assert paths[1]['path'] == [clients[0].address, clients[1].address, clients[4].address,
+                                clients[3].address]
+    assert paths[2]['path'] == [clients[0].address, clients[1].address, clients[2].address,
+                                clients[3].address]
+
+    # Fixme: implement channel close events with balance proofs
+    # now close all channels
+    for channel_id, (
+        p1_index,
+        p1,
+        p1_deposit,
+        p1_transferred_amount,
+        p1_fee,
+        p2_index,
+        p2,
+        p2_deposit,
+        p2_transferred_amount,
+        p2_fee
+    ) in enumerate(channel_descriptions):
+        pass
+
+    # there should be no channels
+    assert True
```

Comment:
this doesn't do anything.

Reply:
I wrote that to have a reminder inside the code as well.

Reply:
just a TODO comment is ok. leaving assert here is confusing.